apps/agent/app/src/components/Transactions.js

```diff
@@ -0,0 +1,476 @@
+              <InnerEntryColumn compactMode={compactMode} theme={theme}>
+                To{' '}
+                {!to ? (
+                  <AppBadge
+                    appAddress={agentAddress}
+                    iconSrc={AgentSvg}
+                    label="Agent"
+                  />
+                ) : (
+                  <LocalIdentityBadge entity={to || 'Agent'} badgeOnly={!to} />
+                )}
+              </InnerEntryColumn>
+              <AmountColumn
+                amountColor={amountColor}
+                compactMode={compactMode}
+                theme={theme}
+              >
+                {compactMode && <span className="amount-label">Amount</span>}
+                <span className="token-amount">
+                  {' '}
+                  {formattedAmount} {symbol}
+                </span>
+              </AmountColumn>
+            </div>
+          )
+        })
+        return compactMode ? (
+          <div
+            css={`
+              width: 100%;
+            `}
+          >
+            {transfers}
+          </div>
+        ) : (
+          transfers
+        )
+      }}
+    />
+  )
+})
+
+Transactions.propTypes = {
+  tokens: PropTypes.array.isRequired,
+  transactions: PropTypes.array.isRequired,
+}
+
+const InnerEntryColumn = styled.div`
+  color: ${({ theme }) => theme.surfaceContentSecondary};
+  ${textStyle('label2')};
+  font-weight: 400;
+  display: inline-grid;
+  grid-template-columns: auto 1fr;
+  grid-gap: ${1 * GU}px;
+  align-items: center;
+  width: 200px;
+  ${({ compactMode }) =>
+    compactMode &&
+    `
+      display: flex;
+      flex-direction: row;
+      align-items: space-between;
+      justify-content: space-between;
+      width: 100%;
+      margin-right: ${5 * GU}px;
+      margin-bottom: ${5 * GU}px;
+    `};
+`
+
+const AmountColumn = styled.div`
+  text-align: right;
+  .amount-label {
+    color: ${({ theme }) => theme.surfaceContentSecondary};
+    ${textStyle('label2')};
+    font-weight: 400;
+  }
+  .token-amount {
+    color: ${({ amountColor }) => amountColor};
+  }
+  ${({ compactMode }) =>
+    compactMode &&
+    `
+      display: flex;
+      display: flex;
+      flex-direction: row;
+      align-items: space-between;
+      justify-content: space-between;
+      width: 100%;
+      margin-right: ${5 * GU}px;
+    `}
+`
+
+function ContextMenuItemCustomLabel({ entity, theme }) {
+  const [label, showLocalIdentityModal] = useIdentity(entity)
+  const handleEditLabel = useCallback(() => showLocalIdentityModal(entity))
+
+  return (
+    <ContextMenuItem onClick={handleEditLabel}>
+      <IconLabel
+        css={`
+          color: ${theme.surfaceContentSecondary};
+        `}
+      />
+      <span
+        css={`
+          margin-left: ${1 * GU}px;
+        `}
+      >
+        {label ? 'Edit' : 'Add'} custom label
+      </span>
+    </ContextMenuItem>
+  )
+}
+
+function ContextMenuViewTransaction({ transactionHash, network, theme }) {
```

Comment:
Something I just noticed: we probably want to use `useTheme()` inside of this component, rather than having it as a prop.

Reply:
Aaah makes sense!